net-vpn/tor/patches/tor-0.3.3.9.patchset

```diff
@@ -35,5 +35,45 @@ index 9c88d37..74f2e48 100644
   * the same IP address and port combination. Otherwise, return 0.
   */
 -- 
-2.7.0
+2.16.4
+
+
+From eb1ea4d7f8d8827f3f5720016f4e23e65479bd45 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Zolt=C3=A1n=20Mizsei?= <zmizsei@extrowerk.com>
+Date: Sun, 2 Sep 2018 12:37:38 +0200
+Subject: Config folder path
+
+
+diff --git a/src/or/config.c b/src/or/config.c
+index cf2f584..1b60026 100644
+--- a/src/or/config.c
++++ b/src/or/config.c
+@@ -5024,7 +5024,11 @@ find_torrc_filename(config_line_t *cmd_arg,
+ #ifndef _WIN32
+       char *fn = NULL;
+       if (!defaults_file) {
+-        fn = expand_filename("~/.torrc");
++        #ifndef __HAIKU__
++          fn = expand_filename("~/.torrc");
++        #else
++          fn = expand_filename("~/config/settings/tor/torrc");
++        #endif
+       }
+       if (fn) {
+         file_status_t hmst = file_status(fn);
+@@ -7749,7 +7753,11 @@ get_data_directory(const char *val)
+ #else /* !(defined(_WIN32)) */
+   const char *d = val;
+   if (!d)
+-    d = "~/.tor";
++    #ifndef __HAIKU__
++      d = "~/.tor";
++    #else
++      d = "~/config/settings/tor";
```

Comment:
IMO `B_USER_VAR_DIRECTORY` might be better suited for storing data from applications.

Reply:
same logic comment as about...  change to ```ifdef __HAIKU__``` and if not let the else be the default.